### PR TITLE
RUN-3503 add system.readRegistryValue

### PR DESCRIPTION
**Pull request by HarsimranSingh**
[Adapter PR](https://github.com/openfin/javascript-adapter/pull/350)

[Win 10 test results (failed passed in isolation)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59fa2bbdaf26a25be2ffc9dc)

[New Test REG_BINARY](https://testing-dashboard.openfin.co/#/app/tests/59ef9a8daf26a25be2ffc948/edit)
[New Test REG_DWORD](https://testing-dashboard.openfin.co/#/app/tests/59ef9909af26a25be2ffc945/edit)
[New Test REG_EXPAND_SZ](https://testing-dashboard.openfin.co/#/app/tests/59ef9b2baf26a25be2ffc949/edit)
[New Test REG_MULTI_SZ](https://testing-dashboard.openfin.co/#/app/tests/59ef9991af26a25be2ffc947/edit)
[New Test REG_SZ](https://testing-dashboard.openfin.co/#/app/tests/59ef990caf26a25be2ffc946/edit)

### CODE REVIEW

**Comment by wenjunche** [comment]
👍

File: src/browser/api_protocol/api_handlers/system.js [hunk 1]
```diff
@@ -80,6 +81,19 @@ function SystemApiHandler() {
         return e !== undefined && e.constructor === Error;
     }
 
+    function readRegistryValue(identity, message, ack, nack) {
+        const { payload: { rootKey, subkey, value } } = message;
+        const response = System.readRegistryValue(rootKey, subkey, value);
+
+        if (_.isEmpty(response)) {
+            nack(new Error('There was an error reading the registry.'));
```
Comment (rdepena) [inline src/browser/api_protocol/api_handlers/system.js]: is there anywhere in the runtime where we can get the actual error that caused an empty read?

**Comment by HarsimranSingh** [comment]
@rdepena forwarding error messages from runtime to the core now.

File: src/browser/api_protocol/api_handlers/system.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ function SystemApiHandler() {
         'open-url-with-browser': openUrlWithBrowser,
         'process-snapshot': processSnapshot,
         'raise-event': raiseEvent,
+        'read-registry-value': readRegistryValue,
         'release-external-process': { apiFunc: releaseExternalProcess, apiPath: '.releaseExternalProcess' },
```
Comment (wenjunche) [inline src/browser/api_protocol/api_handlers/system.js]: missing apiFunc and apiPath

**Comment by HarsimranSingh** [comment]
Updated with permissions stuff @wenjunche

**Comment by wenjunche** [comment]
👍

**Pull request merged**